src/Plugin.ts

```diff
@@ -75,22 +51,38 @@ export class Plugin {
     return null;
   }
 
-  public async saveHar(): Promise<void> {
+  public async saveHar(fileName: string): Promise<void> {
+    this.fileIsDefined(fileName);
+
+    if (!this.connection) {
+      this.logger.err(`Failed to save HAR. First you should start recording.`);
+
+      return null;
+    }
+
     try {
-      const har: Har = new HarBuilder(this.entries).build();
-      await writeFile(this.options.file, JSON.stringify(har, null, 2));
+      await this.fileManager.createIfIsNotExist(resolve(fileName, '..'));
+      const har: string = this.buildHar();
+      await this.fileManager.writeFile(fileName, har);
     } catch (e) {
       this.logger.err(`Failed to save HAR: ${e.message}`);
+    } finally {
+      this.entries = [];
```

Comment:
Are you sure about this? IMO `save()` shouldn't exhibit such side-effect - would be cleaner to have a dedicated `clear()` method that would do that.

Reply:
It's needed to dump network logs. If the end-user don't clear entries it will led to unexpected behaviour like #23. In my opinion it's better solution. Sure, we can provide additional option `preserveLogs` like in chrome. I'd avoid complicating the public interface. Wdyt?

Reply:
Seems reasonable, we can revisit this once there will be a concrete use-case.